core/trino-main/src/main/java/io/trino/sql/gen/columnar/FilterEvaluator.java

```diff
@@ -50,7 +51,14 @@
  * Implementations handle dictionary aware processing through {@link DictionaryAwareColumnarFilter}.
  */
 public sealed interface FilterEvaluator
-        permits AndFilterEvaluator, ColumnarFilterEvaluator, OrFilterEvaluator, PageFilterEvaluator
+        permits
+        AndFilterEvaluator,
+        ColumnarFilterEvaluator,
+        OrFilterEvaluator,
+        PageFilterEvaluator,
+        SelectAllEvaluator,
+        SelectNoneEvaluator,
+        DynamicFilterEvaluator
 {
     SelectionResult evaluate(ConnectorSession session, SelectedPositions activePositions, Page page);
```

Comment:
Why filter evaluator requires `ConnectorSession` rather than just `Session`?

Reply:
That is needed to match existing PageFilter API.
User defined functions can take `ConnectorSession` as an input as that is in the SPI, but `Session` is not.

Reply:
Which connector is used to construct `ConnectorSession`? What is filter is higher-up in the plan and not in SFP

Reply:
You can look at FilterAndProjectOperator/createWorkProcessor/generateFullInvocation, this is all based on existing code.